dex/src/main/scala/com/wavesplatform/dex/Matcher.scala

```diff
@@ -345,34 +395,54 @@ class Matcher(settings: MatcherSettings, gRPCExtensionClient: DEXClient)(implici
 
     log.info(s"Starting matcher on: ${settings.restApi.address}:${settings.restApi.port} ...")
 
-    val startGuard = {
-      wavesBlockchainAsyncClient.getAssetDecimalsForMatchingRules(settings.matchingRules.keySet) flatMap { assetDecimals =>
-        val combinedRoute = new CompositeHttpService(matcherApiTypes, createMatcherApiRoutes(assetDecimals).value, settings.restApi).compositeRoute
-        matcherServerBinding = Await.result(Http().bindAndHandle(combinedRoute, settings.restApi.address, settings.restApi.port), 5.seconds)
-
-        log.info(s"Matcher bound to ${matcherServerBinding.localAddress}")
-        actorSystem.actorOf(
-          ExchangeTransactionBroadcastActor
-            .props(
-              settings.exchangeTransactionBroadcast,
-              time,
-              wavesBlockchainSyncClient.wasForged,
-              wavesBlockchainSyncClient.broadcastTx
-            ),
-          "exchange-transaction-broadcast"
-        )
+    def loadAllKnownAssets(): Future[Unit] = {
+      def parseAsset(label: String)(id: String): Asset =
+        AssetPair.extractAssetId(id).getOrElse(throw new RuntimeException(s"Can't decode asset '$id' from $label"))
+
+      val assetsToLoad = assetPairsDB.all().flatMap(_.assets) ++
+        settings.priceAssets.map(parseAsset("waves.dex.price-assets")) ++ {
+        settings.orderFee match {
+          case x: OrderFeeSettings.FixedSettings => Set(x.defaultAssetId)
+          case _                                 => Set.empty
+        }
+      } ++
+        settings.matchingRules.keySet.flatMap(_.assets) ++
+        settings.blacklistedAssets.map(parseAsset("waves.dex.blacklisted-assets"))
+
+      Future
+        .traverse(assetsToLoad) { asset =>
+          getDecimals(assetsDB, wavesBlockchainAsyncClient.assetDescription)(asset).value
+        }
```

Comment:
Maybe this also can be incapsulated in the dedicated class

Reply:
Probably, we it'd better to move the logic of assets' retrieving from settings. Other lines should stay here IMO, because they depend on incapsulated field of Matcher.